corehq/form_processor/backends/sql/processor.py

```diff
@@ -171,7 +172,12 @@ def get_cases_from_forms(case_db, xforms):
             for xform in xforms:
                 if xform.is_deprecated:
                     deprecated_form = xform
-                affected_cases.update(case_update.id for case_update in get_case_updates(xform))
+                if not (xform.is_deprecated and xform.problem):
+                    # don't process deprecatd forms which have errors.
+                    # see http://manage.dimagi.com/default.asp?243382 for context.
+                    # note that we have to use .problem instead of .is_error because applying
+                    # the state=DEPRECATED overrides state=ERROR
```

Comment:
ok, i think I might punt that for now. actually had to fall back to `.problem` on the couch side too because `is_error` just checks the `doc_type != 'XFormInstance'` which is also too broad